File: mobile/android/UI/SupportMainInterface.kt
package com.aladdin.mobile.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlinx.coroutines.*

// MARK: - Support Main Interface для Android
class SupportMainInterface : AppCompatActivity() {
    
    // UI Components
    private lateinit var scrollView: ScrollView
    private lateinit var contentLayout: LinearLayout
    private lateinit var headerCard: LinearLayout
    private lateinit var titleText: TextView
    private lateinit var subtitleText: TextView
    private lateinit var statusView: LinearLayout
    private lateinit var statusIndicator: View
    private lateinit var statusText: TextView
    
    // Quick Actions
    private lateinit var quickActionsLayout: LinearLayout
    private lateinit var chatButton: SupportActionButton
    private lateinit var faqButton: SupportActionButton
    private lateinit var contactButton: SupportActionButton
    private lateinit var emergencyButton: SupportActionButton
    
    // Categories
    private lateinit var categoriesLayout: LinearLayout
    private lateinit var categoriesTitle: TextView
    
    // Recent Activity
    private lateinit var recentActivityLayout: LinearLayout
    private lateinit var recentActivityTitle: TextView
    private lateinit var recentActivityRecyclerView: RecyclerView
    
    // Properties
    private val supportAPI = UnifiedSupportAPIManager()
    private val scope = CoroutineScope(Dispatchers.Main + SupervisorJob())
    private var recentTickets = listOf<SupportTicket>()
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_support_main)
        
        setupUI()
        setupRecyclerView()
        loadData()
    }
    
    private fun setupUI() {
        // Find views
        scrollView = findViewById(R.id.scrollView)
        contentLayout = findViewById(R.id.contentLayout)
        headerCard = findViewById(R.id.headerCard)
        titleText = findViewById(R.id.titleText)
        subtitleText = findViewById(R.id.subtitleText)
        statusView = findViewById(R.id.statusView)
        statusIndicator = findViewById(R.id.statusIndicator)
        statusText = findViewById(R.id.statusText)
        
        quickActionsLayout = findViewById(R.id.quickActionsLayout)
        chatButton = findViewById(R.id.chatButton)
        faqButton = findViewById(R.id.faqButton)
        contactButton = findViewById(R.id.contactButton)
        emergencyButton = findViewById(R.id.emergencyButton)
        
        categoriesLayout = findViewById(R.id.categoriesLayout)
        categoriesTitle = findViewById(R.id.categoriesTitle)
        
        recentActivityLayout = findViewById(R.id.recentActivityLayout)
        recentActivityTitle = findViewById(R.id.recentActivityTitle)
        recentActivityRecyclerView = findViewById(R.id.recentActivityRecyclerView)
        
        // Setup appearance
        setupAppearance()
        setupQuickActions()
        setupCategories()
    }
    
    private fun setupAppearance() {
        // Header
        titleText.text = "AI Помощник ALADDIN"
        titleText.setTextColor(StormSkyColors.textPrimary)
        
        subtitleText.text = "Ваш персональный помощник по безопасности"
        subtitleText.setTextColor(StormSkyColors.textSecondary)
        
        // Status
        statusIndicator.setBackgroundColor(StormSkyColors.success)
        statusText.text = "Онлайн"
        statusText.setTextColor(StormSkyColors.textPrimary)
        
        // Categories title
        categoriesTitle.text = "Категории поддержки"
        categoriesTitle.setTextColor(StormSkyColors.textPrimary)
        
        // Recent activity title
        recentActivityTitle.text = "Недавние обращения"
        recentActivityTitle.setTextColor(StormSkyColors.textPrimary)
    }
    
    private fun setupQuickActions() {
        // Chat Button
        chatButton.configure(
            title = "💬 Чат с AI",
            subtitle = "Задайте вопрос помощнику",
            action = { openChat() }
        )
        
        // FAQ Button
        faqButton.configure(
            title = "❓ Частые вопросы",
            subtitle = "Быстрые ответы на популярные вопросы",
            action = { openFAQ() }
        )
        
        // Contact Button
        contactButton.configure(
            title = "📞 Связаться с нами",
            subtitle = "Техническая поддержка и консультации",
            action = { openContact() }
        )
        
        // Emergency Button
        emergencyButton.configure(
            title = "🚨 Экстренная помощь",
            subtitle = "Критические ситуации и срочная поддержка",
            action = { openEmergency() }
        )
        emergencyButton.setEmergencyStyle()
    }
    
    private fun setupCategories() {
        val categories = listOf(
            Triple("🛡️ Безопасность", "security", "Защита семьи и устройств"),
            Triple("👨‍👩‍👧‍👦 Семья", "family", "Родительский контроль и детская безопасность"),
            Triple("🔧 Настройки", "settings", "Конфигурация приложения"),
            Triple("💳 Платежи", "payments", "Подписки и оплата"),
            Triple("📱 Устройства", "devices", "Подключение и синхронизация"),
            Triple("🔐 Безопасность данных", "privacy", "Конфиденциальность и шифрование")
        )
        
        categories.forEach { (title, category, description) ->
            val categoryView = SupportCategoryView(this).apply {
                configure(title, description, category)
                setOnClickListener { openCategory(category) }
            }
            categoriesLayout.addView(categoryView)
        }
    }
    
    private fun setupRecyclerView() {
        recentActivityRecyclerView.layoutManager = LinearLayoutManager(this)
        recentActivityRecyclerView.adapter = SupportTicketAdapter(recentTickets) { ticket ->
            openTicketDetails(ticket)
        }
    }
    
    private fun loadData() {
        loadRecentTickets()
        checkSupportStatus()
    }
    
    private fun loadRecentTickets() {
        scope.launch {
            try {
                val tickets = withContext(Dispatchers.IO) {
                    supportAPI.getRecentTickets()
                }
                recentTickets = tickets
                recentActivityRecyclerView.adapter = SupportTicketAdapter(tickets) { ticket ->
                    openTicketDetails(ticket)
                }
            } catch (e: Exception) {
                // Handle error
            }
        }
    }
    
    private fun checkSupportStatus() {
        scope.launch {
            try {
                val status = withContext(Dispatchers.IO) {
                    supportAPI.getSupportStatus()
                }
                updateStatusIndicator(status)
            } catch (e: Exception) {
                // Handle error
            }
        }
    }
    
    private fun updateStatusIndicator(status: SupportStatus) {
        when (status) {
            SupportStatus.ONLINE -> {
                statusIndicator.setBackgroundColor(StormSkyColors.success)
                statusText.text = "Онлайн"
            }
            SupportStatus.BUSY -> {
                statusIndicator.setBackgroundColor(StormSkyColors.warning)
                statusText.text = "Занят"
            }
            SupportStatus.OFFLINE -> {
                statusIndicator.setBackgroundColor(StormSkyColors.error)
                statusText.text = "Офлайн"
            }
        }
    }
    
    // MARK: - Actions
    private fun openChat() {
        val intent = android.content.Intent(this, SupportChatInterface::class.java)
        startActivity(intent)
    }
    
    private fun openFAQ() {
        val intent = android.content.Intent(this, SupportFAQActivity::class.java)
        startActivity(intent)
    }
    
    private fun openContact() {
        val intent = android.content.Intent(this, SupportContactActivity::class.java)
        startActivity(intent)
    }
    
    private fun openEmergency() {
        val intent = android.content.Intent(this, SupportEmergencyActivity::class.java)
        startActivity(intent)
    }
    
    private fun openCategory(category: String) {
        val intent = android.content.Intent(this, SupportCategoryActivity::class.java)
        intent.putExtra("category", category)
        startActivity(intent)
    }
    
    private fun openTicketDetails(ticket: SupportTicket) {
        val intent = android.content.Intent(this, SupportTicketDetailsActivity::class.java)
        intent.putExtra("ticket_id", ticket.id)
        startActivity(intent)
    }
    
    override fun onDestroy() {
        super.onDestroy()
        scope.cancel()
    }
}

// MARK: - Support Action Button
class SupportActionButton : LinearLayout {
    private val iconText: TextView
    private val titleText: TextView
    private val subtitleText: TextView
    private var action: (() -> Unit)? = null
    
    constructor(context: android.content.Context) : super(context) {
        iconText = TextView(context)
        titleText = TextView(context)
        subtitleText = TextView(context)
        init()
    }
    
    constructor(context: android.content.Context, attrs: android.util.AttributeSet?) : super(context, attrs) {
        iconText = TextView(context)
        titleText = TextView(context)
        subtitleText = TextView(context)
        init()
    }
    
    private fun init() {
        orientation = HORIZONTAL
        setPadding(16, 16, 16, 16)
        setBackgroundColor(StormSkyColors.backgroundSecondary)
        background = createRoundedBackground(StormSkyColors.backgroundSecondary)
        
        // Icon
        iconText.textSize = 24f
        iconText.textAlignment = android.view.View.TEXT_ALIGNMENT_CENTER
        
        // Title
        titleText.textSize = 18f
        titleText.setTextColor(StormSkyColors.textPrimary)
        titleText.typeface = android.graphics.Typeface.DEFAULT_BOLD
        
        // Subtitle
        subtitleText.textSize = 14f
        subtitleText.setTextColor(StormSkyColors.textSecondary)
        subtitleText.maxLines = 2
        
        // Layout
        val iconParams = LayoutParams(120, LayoutParams.WRAP_CONTENT).apply {
            gravity = android.view.Gravity.CENTER_VERTICAL
        }
        addView(iconText, iconParams)
        
        val textLayout = LinearLayout(context).apply {
            orientation = VERTICAL
            layoutParams = LayoutParams(0, LayoutParams.WRAP_CONTENT, 1f).apply {
                leftMargin = 12
            }
        }
        
        textLayout.addView(titleText)
        textLayout.addView(subtitleText)
        addView(textLayout)
        
        // Click listener
        setOnClickListener { action?.invoke() }
    }
    
    fun configure(title: String, subtitle: String, action: () -> Unit) {
        val parts = title.split(" ", limit = 2)
        if (parts.size > 1) {
            iconText.text = parts[0]
            titleText.text = parts[1]
        } else {
            titleText.text = title
        }
        subtitleText.text = subtitle
        this.action = action
    }
    
    fun setEmergencyStyle() {
        setBackgroundColor(StormSkyColors.error)
        background = createRoundedBackground(StormSkyColors.error)
        titleText.setTextColor(android.graphics.Color.WHITE)
        subtitleText.setTextColor(android.graphics.Color.WHITE)
    }
    
    private fun createRoundedBackground(color: Int): android.graphics.drawable.Drawable {
        val drawable = android.graphics.drawable.GradientDrawable()
        drawable.setColor(color)
        drawable.cornerRadius = 16f
        return drawable
    }
}

// MARK: - Support Category View
class SupportCategoryView : LinearLayout {
    private val titleText: TextView
    private val descriptionText: TextView
    private val arrowIcon: TextView
    
    constructor(context: android.content.Context) : super(context) {
        titleText = TextView(context)
        descriptionText = TextView(context)
        arrowIcon = TextView(context)
        init()
    }
    
    constructor(context: android.content.Context, attrs: android.util.AttributeSet?) : super(context, attrs) {
        titleText = TextView(context)
        descriptionText = TextView(context)
        arrowIcon = TextView(context)
        init()
    }
    
    private fun init() {
        orientation = HORIZONTAL
        setPadding(16, 12, 16, 12)
        setBackgroundColor(StormSkyColors.backgroundSecondary)
        background = createRoundedBackground(StormSkyColors.backgroundSecondary)
        
        // Title
        titleText.textSize = 16f
        titleText.setTextColor(StormSkyColors.textPrimary)
        titleText.typeface = android.graphics.Typeface.DEFAULT_BOLD
        
        // Description
        descriptionText.textSize = 14f
        descriptionText.setTextColor(StormSkyColors.textSecondary)
        descriptionText.maxLines = 2
        
        // Arrow
        arrowIcon.text = ">"
        arrowIcon.textSize = 18f
        arrowIcon.setTextColor(StormSkyColors.accent)
        
        // Layout
        val textLayout = LinearLayout(context).apply {
            orientation = VERTICAL
            layoutParams = LayoutParams(0, LayoutParams.WRAP_CONTENT, 1f)
        }
        
        textLayout.addView(titleText)
        textLayout.addView(descriptionText)
        addView(textLayout)
        
        val arrowParams = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT).apply {
            gravity = android.view.Gravity.CENTER_VERTICAL
            leftMargin = 8
        }
        addView(arrowIcon, arrowParams)
    }
    
    fun configure(title: String, description: String, category: String) {
        titleText.text = title
        descriptionText.text = description
    }
    
    private fun createRoundedBackground(color: Int): android.graphics.drawable.Drawable {
        val drawable = android.graphics.drawable.GradientDrawable()
        drawable.setColor(color)
        drawable.cornerRadius = 12f
        return drawable
    }
}

// MARK: - Support Ticket Adapter
class SupportTicketAdapter(
    private val tickets: List<SupportTicket>,
    private val onTicketClick: (SupportTicket) -> Unit
) : RecyclerView.Adapter<SupportTicketAdapter.TicketViewHolder>() {
    
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TicketViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_support_ticket, parent, false)
        return TicketViewHolder(view)
    }
    
    override fun onBindViewHolder(holder: TicketViewHolder, position: Int) {
        holder.bind(tickets[position])
    }
    
    override fun getItemCount(): Int = tickets.size
    
    class TicketViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val titleText: TextView = itemView.findViewById(R.id.ticketTitle)
        private val statusText: TextView = itemView.findViewById(R.id.ticketStatus)
        private val dateText: TextView = itemView.findViewById(R.id.ticketDate)
        private val lastMessageText: TextView = itemView.findViewById(R.id.ticketLastMessage)
        
        fun bind(ticket: SupportTicket) {
            titleText.text = ticket.title
            statusText.text = ticket.status.name
            dateText.text = formatDate(ticket.createdAt)
            lastMessageText.text = ticket.lastMessage ?: "Нет сообщений"
            
            // Set status color
            when (ticket.status) {
                TicketStatus.OPEN -> statusText.setTextColor(StormSkyColors.warning)
                TicketStatus.IN_PROGRESS -> statusText.setTextColor(StormSkyColors.accent)
                TicketStatus.RESOLVED -> statusText.setTextColor(StormSkyColors.success)
                TicketStatus.CLOSED -> statusText.setTextColor(StormSkyColors.textSecondary)
            }
        }
        
        private fun formatDate(date: Date): String {
            val formatter = java.text.SimpleDateFormat("dd.MM.yyyy", Locale.getDefault())
            return formatter.format(date)
        }
    }
}

// MARK: - Support Models
data class SupportTicket(
    val id: String,
    val title: String,
    val status: TicketStatus,
    val category: String,
    val createdAt: Date,
    val lastMessage: String?
)

enum class TicketStatus {
    OPEN, IN_PROGRESS, RESOLVED, CLOSED
}

enum class SupportStatus {
    ONLINE, BUSY, OFFLINE
}

File: mobile/ios/UI/SupportMainInterface.swift
import UIKit
import Combine

// MARK: - Support Main Interface для iOS
class SupportMainInterface: UIViewController {
    
    // MARK: - UI Components
    private let scrollView = UIScrollView()
    private let contentView = UIView()
    private let headerView = UIView()
    private let titleLabel = UILabel()
    private let subtitleLabel = UILabel()
    private let statusView = UIView()
    private let statusLabel = UILabel()
    private let statusIndicator = UIView()
    
    // Quick Actions
    private let quickActionsStackView = UIStackView()
    private let chatButton = SupportActionButton()
    private let faqButton = SupportActionButton()
    private let contactButton = SupportActionButton()
    private let emergencyButton = SupportActionButton()
    
    // Support Categories
    private let categoriesStackView = UIStackView()
    private let categoriesLabel = UILabel()
    
    // Recent Activity
    private let recentActivityView = UIView()
    private let recentActivityLabel = UILabel()
    private let recentActivityTableView = UITableView()
    
    // Properties
    private var cancellables = Set<AnyCancellable>()
    private let supportAPI = UnifiedSupportAPIManager()
    private var recentTickets: [SupportTicket] = []
    
    // MARK: - Lifecycle
    override func viewDidLoad() {
        super.viewDidLoad()
        setupUI()
        setupConstraints()
        setupBindings()
        loadData()
    }
    
    // MARK: - UI Setup
    private func setupUI() {
        view.backgroundColor = StormSkyColors.backgroundPrimary
        
        // Navigation
        navigationItem.title = "Поддержка"
        navigationItem.rightBarButtonItem = UIBarButtonItem(
            title: "Настройки",
            style: .plain,
            target: self,
            action: #selector(settingsButtonTapped)
        )
        
        // Header View
        setupHeaderView()
        
        // Quick Actions
        setupQuickActions()
        
        // Categories
        setupCategories()
        
        // Recent Activity
        setupRecentActivity()
        
        // Add subviews
        view.addSubview(scrollView)
        scrollView.addSubview(contentView)
        contentView.addSubview(headerView)
        contentView.addSubview(quickActionsStackView)
        contentView.addSubview(categoriesStackView)
        contentView.addSubview(recentActivityView)
    }
    
    private func setupHeaderView() {
        headerView.backgroundColor = StormSkyColors.backgroundSecondary
        headerView.layer.cornerRadius = 16
        headerView.layer.borderWidth = 1
        headerView.layer.borderColor = StormSkyColors.accent.cgColor
        
        // Title
        titleLabel.text = "AI Помощник ALADDIN"
        titleLabel.font = UIFont.boldSystemFont(ofSize: 24)
        titleLabel.textColor = StormSkyColors.textPrimary
        titleLabel.textAlignment = .center
        
        // Subtitle
        subtitleLabel.text = "Ваш персональный помощник по безопасности"
        subtitleLabel.font = UIFont.systemFont(ofSize: 16)
        subtitleLabel.textColor = StormSkyColors.textSecondary
        subtitleLabel.textAlignment = .center
        subtitleLabel.numberOfLines = 0
        
        // Status
        statusIndicator.backgroundColor = StormSkyColors.success
        statusIndicator.layer.cornerRadius = 6
        statusIndicator.layer.borderWidth = 2
        statusIndicator.layer.borderColor = StormSkyColors.accent.cgColor
        
        statusLabel.text = "Онлайн"
        statusLabel.font = UIFont.systemFont(ofSize: 14)
        statusLabel.textColor = StormSkyColors.textPrimary
        
        [titleLabel, subtitleLabel, statusView].forEach {
            $0.translatesAutoresizingMaskIntoConstraints = false
            headerView.addSubview($0)
        }
        
        [statusIndicator, statusLabel].forEach {
            $0.translatesAutoresizingMaskIntoConstraints = false
            statusView.addSubview($0)
        }
    }
    
    private func setupQuickActions() {
        quickActionsStackView.axis = .vertical
        quickActionsStackView.spacing = 16
        quickActionsStackView.distribution = .fillEqually
        
        // Chat Button
        chatButton.configure(
            title: "💬 Чат с AI",
            subtitle: "Задайте вопрос помощнику",
            action: { [weak self] in self?.openChat() }
        )
        
        // FAQ Button
        faqButton.configure(
            title: "❓ Частые вопросы",
            subtitle: "Быстрые ответы на популярные вопросы",
            action: { [weak self] in self?.openFAQ() }
        )
        
        // Contact Button
        contactButton.configure(
            title: "📞 Связаться с нами",
            subtitle: "Техническая поддержка и консультации",
            action: { [weak self] in self?.openContact() }
        )
        
        // Emergency Button
        emergencyButton.configure(
            title: "🚨 Экстренная помощь",
            subtitle: "Критические ситуации и срочная поддержка",
            action: { [weak self] in self?.openEmergency() }
        )
        emergencyButton.setEmergencyStyle()
        
        [chatButton, faqButton, contactButton, emergencyButton].forEach {
            quickActionsStackView.addArrangedSubview($0)
        }
    }
    
    private func setupCategories() {
        categoriesLabel.text = "Категории поддержки"
        categoriesLabel.font = UIFont.boldSystemFont(ofSize: 20)
        categoriesLabel.textColor = StormSkyColors.textPrimary
        
        categoriesStackView.axis = .vertical
        categoriesStackView.spacing = 12
        
        let categories = [
            ("🛡️ Безопасность", "security", "Защита семьи и устройств"),
            ("👨‍👩‍👧‍👦 Семья", "family", "Родительский контроль и детская безопасность"),
            ("🔧 Настройки", "settings", "Конфигурация приложения"),
            ("💳 Платежи", "payments", "Подписки и оплата"),
            ("📱 Устройства", "devices", "Подключение и синхронизация"),
            ("🔐 Безопасность данных", "privacy", "Конфиденциальность и шифрование")
        ]
        
        categories.forEach { (title, category, description) in
            let categoryView = SupportCategoryView()
            categoryView.configure(title: title, description: description, category: category)
            categoryView.onTap = { [weak self] in
                self?.openCategory(category)
            }
            categoriesStackView.addArrangedSubview(categoryView)
        }
        
        [categoriesLabel, categoriesStackView].forEach {
            $0.translatesAutoresizingMaskIntoConstraints = false
            categoriesStackView.addArrangedSubview($0)
        }
    }
    
    private func setupRecentActivity() {
        recentActivityLabel.text = "Недавние обращения"
        recentActivityLabel.font = UIFont.boldSystemFont(ofSize: 20)
        recentActivityLabel.textColor = StormSkyColors.textPrimary
        
        recentActivityTableView.delegate = self
        recentActivityTableView.dataSource = self
        recentActivityTableView.backgroundColor = .clear
        recentActivityTableView.separatorStyle = .none
        recentActivityTableView.register(SupportTicketCell.self, forCellReuseIdentifier: "TicketCell")
        recentActivityTableView.isScrollEnabled = false
        
        [recentActivityLabel, recentActivityTableView].forEach {
            $0.translatesAutoresizingMaskIntoConstraints = false
            recentActivityView.addSubview($0)
        }
    }
    
    private func setupConstraints() {
        [scrollView, contentView, headerView, titleLabel, subtitleLabel, statusView, statusIndicator, statusLabel, quickActionsStackView, categoriesStackView, recentActivityView, recentActivityLabel, recentActivityTableView].forEach {
            $0.translatesAutoresizingMaskIntoConstraints = false
        }
        
        NSLayoutConstraint.activate([
            // Scroll View
            scrollView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
            scrollView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
            scrollView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
            scrollView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
            
            // Content View
            contentView.topAnchor.constraint(equalTo: scrollView.topAnchor),
            contentView.leadingAnchor.constraint(equalTo: scrollView.leadingAnchor),
            contentView.trailingAnchor.constraint(equalTo: scrollView.trailingAnchor),
            contentView.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor),
            contentView.widthAnchor.constraint(equalTo: scrollView.widthAnchor),
            
            // Header View
            headerView.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 16),
            headerView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
            headerView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
            
            // Title
            titleLabel.topAnchor.constraint(equalTo: headerView.topAnchor, constant: 20),
            titleLabel.leadingAnchor.constraint(equalTo: headerView.leadingAnchor, constant: 20),
            titleLabel.trailingAnchor.constraint(equalTo: headerView.trailingAnchor, constant: -20),
            
            // Subtitle
            subtitleLabel.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: 8),
            subtitleLabel.leadingAnchor.constraint(equalTo: headerView.leadingAnchor, constant: 20),
            subtitleLabel.trailingAnchor.constraint(equalTo: headerView.trailingAnchor, constant: -20),
            
            // Status View
            statusView.topAnchor.constraint(equalTo: subtitleLabel.bottomAnchor, constant: 16),
            statusView.centerXAnchor.constraint(equalTo: headerView.centerXAnchor),
            statusView.bottomAnchor.constraint(equalTo: headerView.bottomAnchor, constant: -20),
            statusView.heightAnchor.constraint(equalToConstant: 24),
            
            // Status Indicator
            statusIndicator.leadingAnchor.constraint(equalTo: statusView.leadingAnchor),
            statusIndicator.centerYAnchor.constraint(equalTo: statusView.centerYAnchor),
            statusIndicator.widthAnchor.constraint(equalToConstant: 12),
            statusIndicator.heightAnchor.constraint(equalToConstant: 12),
            
            // Status Label
            statusLabel.leadingAnchor.constraint(equalTo: statusIndicator.trailingAnchor, constant: 8),
            statusLabel.trailingAnchor.constraint(equalTo: statusView.trailingAnchor),
            statusLabel.centerYAnchor.constraint(equalTo: statusView.centerYAnchor),
            
            // Quick Actions
            quickActionsStackView.topAnchor.constraint(equalTo: headerView.bottomAnchor, constant: 24),
            quickActionsStackView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
            quickActionsStackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
            
            // Categories
            categoriesStackView.topAnchor.constraint(equalTo: quickActionsStackView.bottomAnchor, constant: 32),
            categoriesStackView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
            categoriesStackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
            
            // Recent Activity
            recentActivityView.topAnchor.constraint(equalTo: categoriesStackView.bottomAnchor, constant: 32),
            recentActivityView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
            recentActivityView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
            recentActivityView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -32),
            
            // Recent Activity Label
            recentActivityLabel.topAnchor.constraint(equalTo: recentActivityView.topAnchor),
            recentActivityLabel.leadingAnchor.constraint(equalTo: recentActivityView.leadingAnchor),
            recentActivityLabel.trailingAnchor.constraint(equalTo: recentActivityView.trailingAnchor),
            
            // Recent Activity Table View
            recentActivityTableView.topAnchor.constraint(equalTo: recentActivityLabel.bottomAnchor, constant: 16),
            recentActivityTableView.leadingAnchor.constraint(equalTo: recentActivityView.leadingAnchor),
            recentActivityTableView.trailingAnchor.constraint(equalTo: recentActivityView.trailingAnchor),
            recentActivityTableView.bottomAnchor.constraint(equalTo: recentActivityView.bottomAnchor),
            recentActivityTableView.heightAnchor.constraint(equalToConstant: 200)
        ])
    }
    
    private func setupBindings() {
        // Load data when view appears
        NotificationCenter.default.publisher(for: UIApplication.didBecomeActiveNotification)
            .sink { [weak self] _ in
                self?.loadData()
            }
            .store(in: &cancellables)
    }
    
    // MARK: - Data Loading
    private func loadData() {
        loadRecentTickets()
        checkSupportStatus()
    }
    
    private func loadRecentTickets() {
        supportAPI.getRecentTickets()
            .receive(on: DispatchQueue.main)
            .sink(
                receiveCompletion: { _ in },
                receiveValue: { [weak self] tickets in
                    self?.recentTickets = tickets
                    self?.recentActivityTableView.reloadData()
                }
            )
            .store(in: &cancellables)
    }
    
    private func checkSupportStatus() {
        supportAPI.getSupportStatus()
            .receive(on: DispatchQueue.main)
            .sink(
                receiveCompletion: { _ in },
                receiveValue: { [weak self] status in
                    self?.updateStatusIndicator(status)
                }
            )
            .store(in: &cancellables)
    }
    
    private func updateStatusIndicator(_ status: SupportStatus) {
        switch status {
        case .online:
            statusIndicator.backgroundColor = StormSkyColors.success
            statusLabel.text = "Онлайн"
        case .busy:
            statusIndicator.backgroundColor = StormSkyColors.warning
            statusLabel.text = "Занят"
        case .offline:
            statusIndicator.backgroundColor = StormSkyColors.error
            statusLabel.text = "Офлайн"
        }
    }
    
    // MARK: - Actions
    @objc private func settingsButtonTapped() {
        let settingsVC = SupportSettingsViewController()
        let navController = UINavigationController(rootViewController: settingsVC)
        present(navController, animated: true)
    }
    
    private func openChat() {
        let chatVC = SupportChatInterface()
        let navController = UINavigationController(rootViewController: chatVC)
        present(navController, animated: true)
    }
    
    private func openFAQ() {
        let faqVC = SupportFAQViewController()
        let navController = UINavigationController(rootViewController: faqVC)
        present(navController, animated: true)
    }
    
    private func openContact() {
        let contactVC = SupportContactViewController()
        let navController = UINavigationController(rootViewController: contactVC)
        present(navController, animated: true)
    }
    
    private func openEmergency() {
        let emergencyVC = SupportEmergencyViewController()
        let navController = UINavigationController(rootViewController: emergencyVC)
        present(navController, animated: true)
    }
    
    private func openCategory(_ category: String) {
        let categoryVC = SupportCategoryViewController(category: category)
        let navController = UINavigationController(rootViewController: categoryVC)
        present(navController, animated: true)
    }
}

// MARK: - Table View Data Source & Delegate
extension SupportMainInterface: UITableViewDataSource, UITableViewDelegate {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return min(recentTickets.count, 3) // Show max 3 recent tickets
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "TicketCell", for: indexPath) as! SupportTicketCell
        cell.configure(with: recentTickets[indexPath.row])
        return cell
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 60
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        let ticket = recentTickets[indexPath.row]
        openTicketDetails(ticket)
    }
    
    private func openTicketDetails(_ ticket: SupportTicket) {
        let detailsVC = SupportTicketDetailsViewController(ticket: ticket)
        let navController = UINavigationController(rootViewController: detailsVC)
        present(navController, animated: true)
    }
}

// MARK: - Support Action Button
class SupportActionButton: UIView {
    private let titleLabel = UILabel()
    private let subtitleLabel = UILabel()
    private let iconLabel = UILabel()
    private var action: (() -> Void)?
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        setupUI()
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        setupUI()
    }
    
    private func setupUI() {
        backgroundColor = StormSkyColors.backgroundSecondary
        layer.cornerRadius = 16
        layer.borderWidth = 1
        layer.borderColor = StormSkyColors.accent.cgColor
        
        // Icon
        iconLabel.font = UIFont.systemFont(ofSize: 24)
        iconLabel.textAlignment = .center
        
        // Title
        titleLabel.font = UIFont.boldSystemFont(ofSize: 18)
        titleLabel.textColor = StormSkyColors.textPrimary
        
        // Subtitle
        subtitleLabel.font = UIFont.systemFont(ofSize: 14)
        subtitleLabel.textColor = StormSkyColors.textSecondary
        subtitleLabel.numberOfLines = 0
        
        [iconLabel, titleLabel, subtitleLabel].forEach {
            $0.translatesAutoresizingMaskIntoConstraints = false
            addSubview($0)
        }
        
        NSLayoutConstraint.activate([
            iconLabel.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 16),
            iconLabel.centerYAnchor.constraint(equalTo: centerYAnchor),
            iconLabel.widthAnchor.constraint(equalToConstant: 40),
            
            titleLabel.leadingAnchor.constraint(equalTo: iconLabel.trailingAnchor, constant: 12),
            titleLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -16),
            titleLabel.topAnchor.constraint(equalTo: topAnchor, constant: 16),
            
            subtitleLabel.leadingAnchor.constraint(equalTo: titleLabel.leadingAnchor),
            subtitleLabel.trailingAnchor.constraint(equalTo: titleLabel.trailingAnchor),
            subtitleLabel.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: 4),
            subtitleLabel.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -16)
        ])
        
        // Tap gesture
        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(buttonTapped))
        addGestureRecognizer(tapGesture)
    }
    
    func configure(title: String, subtitle: String, action: @escaping () -> Void) {
        let components = title.components(separatedBy: " ")
        if let emoji = components.first {
            iconLabel.text = emoji
            titleLabel.text = components.dropFirst().joined(separator: " ")
        } else {
            titleLabel.text = title
        }
        subtitleLabel.text = subtitle
        self.action = action
    }
    
    func setEmergencyStyle() {
        backgroundColor = StormSkyColors.error.withAlphaComponent(0.1)
        layer.borderColor = StormSkyColors.error.cgColor
        titleLabel.textColor = StormSkyColors.error
    }
    
    @objc private func buttonTapped() {
        action?()
    }
}

// MARK: - Support Models
struct SupportTicket {
    let id: String
    let title: String
    let status: TicketStatus
    let category: String
    let createdAt: Date
    let lastMessage: String?
}

enum TicketStatus {
    case open
    case inProgress
    case resolved
    case closed
}

enum SupportStatus {
    case online
    case busy
    case offline
}

